packages/blade/src/components/Tooltip/_decisions/decisions.md

```diff
@@ -0,0 +1,221 @@
+| placement    | `top, top-start, top-end, left, right, bottom, bottom-start, bottom-end` | `top`       | Placement of tooltip, the tooltip avoid collision with the edge of the screen and flip to the oppisite side even when placement is set. |          |
+| onOpenChange | `({ isOpen }) => void`                                                   | `undefined` | Called when tooltip isOpen state is changed, this can be used to detect when tooltip opens or closed                                    |          |
+
+### Examples:
+
+#### Usage
+
+```js
+// basic example
+<Tooltip content="Amount reversed to customer bank account" placement="top">
+  <IconButton icon={InfoIcon} accessibilityLabel="Refund" />
+</Tooltip>
+```
+
+### Working with non-interactive triggers
+
+To make tooltip apear on hovering over non-interactive elements such as icons, badges, counters etc. We will provide a `TooltipInteractiveWrapper` component which will work as a minimal trigger:
+
+```js
+import { Tooltip, TooltipInteractiveWrapper } from '@razorpay/blade';
+
+// non-interactive element as trigger
+<Tooltip content="Amount reversed to customer bank account">
+  <TooltipInteractiveWrapper>
+    <Counter value={100} />
+  </TooltipInteractiveWrapper>
+</Tooltip>;
+```
+
+> Note: If users don't wrap the non-interactive elements the tooltip won't apear
+
+### Custom Triggers
+
+Users will be able to use their own custom components as triggers:
+
+To make custom elements work the components needs to expose:
+
+- ref
+- Web:
+  - onBlur
+  - onFocus
+  - onMouseLeave
+  - onMouseMove
+  - onPointerDown
+  - onPointerEnter
+- Native:
+  - onTouchStart
+  - onTouchEnd
+
+```jsx
+import type { TooltipTriggerProps } from "@razorpay/blade/components";
+
+type MyCustomButtonProps = { 
+  children: React.ReactNode 
+} & TooltipTriggerProps
+
+const MyCustomButton = React.forwardRef<
+  HTMLDivElement,
+  MyCustomButtonProps
+>(({ children, ...props }, ref) => {
+  return (
+    <button
+      ref={ref}
+      tabIndex={0}
+      // you can either spread the props
+      // or pick the onFocus, onMouse.. etc events and add them individually
+      {...props}
+    >
+       {children}
+    </button>
+  );
+});
+
+// custom trigger
+<Tooltip content="Amount reversed to customer bank account">
+  <MyCustomButton>Click me<MyCustomButton>
+</Tooltip>
+```
+
+## Library
+
+We will be using [FloatingUI](https://floating-ui.com/) to position the tooltip & handle the basic tooltip logic. Floating UI supports both Web & ReactNative.
+
+### Bundle Size
+
+With all the neccessary middlewares and features included for our usecase FloatingUI will be around **10kb** gzip.
+
+Check the [BundleJS Analysis](https://bundlejs.com/?q=%40floating-ui%2Freact&treeshake=%5B%7B%0A++arrow%2Cflip%2CFloatingArrow%2Coffset%2CuseFloating%2CuseFocus%2CuseHover%2CuseInteractions%2CuseTransitionStyles%2C%7D%5D&config=%7B%22esbuild%22%3A%7B%22external%22%3A%5B%22react%22%2C%22react-dom%22%5D%7D%7D)
+
```

Comment:
is this SEO friendly?

Reply:
Didn't understood 🤔 SEO as in? Do you mean tooltip's content should remain on DOM even if it's hidden?